src/crimson/os/seastore/seastore.cc

```diff
@@ -1385,6 +1403,70 @@ SeaStore::tm_ret SeaStore::_setattrs(
     std::move(aset));
 }
 
+SeaStore::tm_ret SeaStore::_rmattr(
+  internal_context_t &ctx,
+  OnodeRef &onode,
+  std::string name)
+{
+  LOG_PREFIX(SeaStore::_rmattr);
+  DEBUGT("onode={}", *ctx.transaction, *onode);
+  auto& layout = onode->get_mutable_layout(*ctx.transaction);
+  if (name == OI_ATTR) {
+    memset(&layout.oi[0], 0, layout.oi_size);
+    layout.oi_size = 0;
+    return tm_iertr::now();
+  } else if (name == SS_ATTR) {
+    memset(&layout.ss[0], 0, layout.ss_size);
+    layout.ss_size = 0;
+    return tm_iertr::now();
+  } else {
+    std::set<std::string> str;
+    str.insert(name);
+    return _omap_rmkeys(ctx, onode, std::move(str));
+  }
+}
+
+SeaStore::tm_ret SeaStore::_rmattrs(
+  internal_context_t &ctx,
+  OnodeRef &onode)
+{
+  LOG_PREFIX(SeaStore::_rmattrs);
+  DEBUGT("onode={}", *ctx.transaction, *onode);
+  auto& layout = onode->get_mutable_layout(*ctx.transaction);
+  memset(&layout.oi[0], 0, layout.oi_size);
+  layout.oi_size = 0;
+  memset(&layout.ss[0], 0, layout.ss_size);
+  layout.ss_size = 0;
+  return _omap_clear(ctx, onode);
+}
+
+SeaStore::tm_ret SeaStore::_omap_clear(
+  internal_context_t &ctx,
+  OnodeRef &onode)
+{
+  LOG_PREFIX(SeaStore::_omap_clear);
+  DEBUGT("onode={}", *ctx.transaction, *onode);
+  auto omap_root = onode->get_layout().omap_root.get(
```

Comment:
_omap_clear here uses the omap_root, which is fine but we invoke it above from rmattrs which should be using the xattr_root -- see _setattrs.